src/modules/launcher/Wox.Infrastructure/Image/WindowsThumbnailProvider.cs

```diff
@@ -125,5 +129,67 @@ private static IntPtr GetHBitmap(string fileName, int width, int height, Thumbna
 
             throw new InvalidComObjectException($"Error while extracting thumbnail for {fileName}", Marshal.GetExceptionForHR((int)hr));
         }
+
+        private static bool reportedAdobeReaderDetected = false; // Keep track if Adobe Reader detection has been logged yet.
+        private static bool reportedErrorInDetectingAdobeReader = false; // Keep track if we reported an exception while trying to detect Adobe Reader yet.
+
+        // We have to evaluate this in real time to not crash, if the user switches to Adobe Reader/Acrobat Pro after starting PT Run.
+        // Adobe registers its thumbnail handler always in "HKCR\Acrobat.Document.*\shellex\{BB2E617C-0920-11d1-9A0B-00C04FC2D6C1}".
+        public static bool DoesPdfUseAcrobatAsProvider()
+        {
+            try
+            {
+                // First check of there is an provider other than Adobe. For example PowerToys.
+                // Generic Guids used by Explorer to identify the configured provider types: {BB2E617C-0920-11d1-9A0B-00C04FC2D6C1} = Image thumbnail; {E357FCCD-A995-4576-B01F-234630154E96} = File thumbnail;
+                RegistryKey key1 = Registry.ClassesRoot.OpenSubKey(".pdf\\shellex\\{E357FCCD-A995-4576-B01F-234630154E96}", false);
+                string value1 = (string)key1?.GetValue(string.Empty);
+                key1?.Close();
+                RegistryKey key2 = Registry.ClassesRoot.OpenSubKey(".pdf\\shellex\\{BB2E617C-0920-11d1-9A0B-00C04FC2D6C1}", false);
+                string value2 = (string)key2?.GetValue(string.Empty);
+                key2?.Close();
+                if (!string.IsNullOrEmpty(value1) || !string.IsNullOrEmpty(value2))
+                {
+                    return false;
```

Comment:
Should we add a small comment like "Other app like PowerToys is registered as provider."?